### PR TITLE
feat(input): add the large size for the ionic theme

### PR DESCRIPTION
**Pull request by brandyscarney**
Issue number: internal

---------

## What is the current behavior?
Input does not have a size property.

## What is the new behavior?

- Adds the `size` property with support for the `"large"` size on the `"ionic"` theme only
- Adds tests for the size property
- Note: the screenshots will not look right until the fill styles are added

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/input/test/size/index.html [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        <div class="grid">
+          <div class="grid-item">
+            <h2>No Fill: No Size</h2>
+            <ion-input label="Email"></ion-input>
+          </div>
+
+          <div class="grid-item">
+            <h2>Solid: No Size</h2>
+            <ion-input fill="solid" label="Email"></ion-input>
+          </div>
```
Comment (thetaPC) [inline core/src/components/input/test/size/index.html:59]: OS doen't have a solid appearance, this can be removed along with the large version.

**Review by thetaPC — CHANGES_REQUESTED** [review]

File: core/src/components/input/input.tsx [hunk 2]
```diff
@@ -251,6 +252,12 @@ export class Input implements ComponentInterface {
    */
   @Prop() step?: string;
 
+  /**
+   * The size of the input. If "large" it will have an increased height. By default the
```
[the remark this answers is not in the document]
Comment (brandyscarney) [reply core/src/components/input/input.tsx]: https://github.com/ionic-team/ionic-framework/pull/29249/commits/f9727438d04c93833de88988625574fe0ce862f3

File: core/src/components/input/test/size/index.html (hunk 1 above)
Comment (brandyscarney) [reply core/src/components/input/test/size/index.html]: Good catch, I was copy/pasting from another test! Fixed in https://github.com/ionic-team/ionic-framework/pull/29249/commits/9f13557c69d067d285429dba5e6f8c8f2a704179

**Review by thetaPC — APPROVED** [review]
LGTM, verified the size increase locally

**Review by sean-perkins — APPROVED** [review]

**Pull request merged**